posthog/api/test/test_feature_flag.py

```diff
@@ -352,6 +367,126 @@ def test_deleting_feature_flag(self):
             },
         )
 
+        historical_flags = HistoricalVersion.objects.filter(team_id=self.team.id, name="FeatureFlag")
+        self.assertEqual(len(historical_flags), 1)
+        self.maxDiff = None
+        self.assertEqual(
+            historical_flags[0].state,
+            {
+                "id": None,
+                "key": "potato",
+                "name": "",
+                "_state": None,
+                "active": True,
+                "deleted": False,
```

Comment:
How come this is false? Do we store the state before the changes... or after the changes?

Reply:
This is testing the delete API call

That property is when we carry out a delete in the UI (which is a soft delete). So this flag hasn't been soft deleted and it's delete property is false but it has been deleted 🤯

Reply:
I'll add a comment here to help the future traveller

Reply:
You added a comment 40 lines above here, I'd nit and add another `//` here as well, as we can't be sure future traveller reads from top to bottom...